primer/src/Primer/Def/Utils.hs

```diff
@@ -26,6 +27,19 @@ nextID (DefAST (ASTDef e t)) =
 nextID (DefPrim _) = 0
 {-# INLINE nextID #-}
 
+-- | Given a 'TypeDef', return its next 'ID'.
+--
+-- Note: do not rely on the implementation of this function, as it may
+-- change in the future.
+nextIDTypeDef :: TypeDef TypeMeta KindMeta -> ID
+nextIDTypeDef (TypeDefAST (ASTTypeDef ps vcs _)) =
+  let pid = foldlOf' (folded % _2 % kindIDs) max minBound ps
+      vcid = foldlOf' (folded % #valConArgs % folded % typeIDs) max minBound vcs
+   in succ $ max pid vcid
+nextIDTypeDef (TypeDefPrim (PrimTypeDef ps _)) =
+  succ $ foldlOf' (folded % _2 % kindIDs) max minBound ps
+{-# INLINE nextIDTypeDef #-}
```

Comment:
I'm intrigued, given that we have never really done any serious low-level optimising, about why we always want to inline this particular? This and the function above are the only two such pragmas in the codebase.

Reply:
I don't recall doing that. 

I don't mind removing it so long as it doesn't adversely affect the benchmarks, but on the other hand, I doubt it's dong any harm, either.